### ISSUE TITLE
Boxes.filter_boxes_by_area: a "zeroed" box exports as a real [0, 0, 1, 1] box in the exclusive modes

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4714]
## Summary

`Boxes.filter_boxes_by_area` documents that an out-of-range box is "replaced with zeros". It zeroes the **stored inclusive** vertices, and every exclusive export (`'xyxy'`, `'xywh'`, `'vertices'`) adds the pixel back, so a filtered box comes out as `[0, 0, 1, 1]`: a real-looking 1x1 box at the origin, indistinguishable from a genuine one, and `to_mask` fills pixel `(0, 0)` for it. Only the `_plus` exports show zeros.

Because the stored form is inclusive, no stored value can export as a zero-area box in an exclusive mode (exclusive width = inclusive width + 1 >= 1), so a caller cannot fix this by writing a different sentinel into `_data`.

## Repro

```python
import torch
from kornia.geometry.boxes import Boxes

b = Boxes.from_tensor(torch.tensor([[[1., 1., 4., 4.], [1., 1., 2., 2.]]]), mode="xyxy")
f = b.filter_boxes_by_area(min_area=2.0)          # drops the second (1x1) box
for m in ("xyxy", "xywh", "vertices", "xyxy_plus", "vertices_plus"):
    print(m, f.to_tensor(m)[0, 1].tolist())
print("to_mask:", f.to_mask(3, 3)[0, 1].tolist())
```

## Observed

```
xyxy [0.0, 0.0, 1.0, 1.0]
xywh [0.0, 0.0, 1.0, 1.0]
vertices [[0.0, 0.0], [1.0, 0.0], [1.0, 1.0], [0.0, 1.0]]
xyxy_plus [0.0, 0.0, 0.0, 0.0]
vertices_plus [[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]]
to_mask: [[1.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]
```

## Expected

A filtered box exports as an unambiguous "no box" in every mode, e.g. all zeros (zero area) in `'xyxy'` / `'xywh'` / `'vertices'` too, and `to_mask` fills nothing for it. That needs either a validity mask carried by `Boxes` (as `_N` already does for trailing padding) and applied on export, or the exclusive storage from #3934.

## Where it shows up

Augmentations call `filter_boxes_by_area` on their outputs, e.g. `RandomMosaic` with `min_bbox_size` (a too-small mosaic box comes back as `[0, 0, 1, 1]` in `bbox_xyxy`). This is the same `[0, 0, 1, 1]` that #4652 called a phantom box; #4679 stops `RandomMosaic` from producing it as *padding* (padding rows are now exported as exact zeros through `_N`), but the filtered-box case goes through this shared `Boxes` code and is left for this issue.

## Classification

Broken output by the method's own documented contract ("replaced with zeros"). Root cause is the inclusive representation, same as #3934; related: #4010 (`compute_area` is `(w - 1) * (h - 1)`), #4009 (exclusive export fed to the inclusive free functions).

## Environment

kornia main @ 088586663, torch 2.14.0, python 3.11, CPU, macOS arm64. Found while implementing the #4652 padding ruling on #4679.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
